### PR TITLE
update links to group() API

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

At https://angular.io/guide/complex-animation-sequences all group() links redirects to https://angular.io/api/forms/FormBuilder#group
Issue Number: 29213

## What is the new behavior?
Updated links to point to the animations groups() API

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by brandonroberts** [comment]
This makes the `group` link stand out from the rest of the links in the list

**Comment by brandonroberts** [comment]
@chembu what about making all the items in the list links. The other option would be too enable some differentiation between API links? @petebacondarwin and @gkalpak would have a better idea on that.

**Comment by petebacondarwin** [comment]
A workaround is to hard-code the HTML:

```
* <code><a href="api/animations/group" class="code-anchor">group()</a></code> runs multiple animation steps in parallel.
```

**Comment by petebacondarwin** [comment]
But a proper solution is to change this style here (https://github.com/angular/angular/blob/master/aio/src/styles/0-base/_typography.scss#L151-L154) to 

```scss
code {
  font-family: $code-font;
  font-size: 90%;
  color: $darkgray;
}
```

Note the addition of the `color: $darkgray`, which overrides the anchor color.

**Comment by chembu** [comment]
> But a proper solution is to change this style here 

@petebacondarwin: Shall I go ahead and update that file as per your suggestion?

**Comment by petebacondarwin** [comment]
Go for it!

File: aio/content/guide/complex-animation-sequences.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Functions that control complex animation sequences are as follows:
 
 * `query()` finds one or more inner HTML elements.
 * `stagger()` applies a cascading delay to animations for multiple elements.
-* `group()` runs multiple animation steps in parallel.
+* [group()](api/animations/group) runs multiple animation steps in parallel.
```
Comment (petebacondarwin) [inline aio/content/guide/complex-animation-sequences.md]: But you still need the backticks for it to be given the code font....

```suggestion
* [`group()`](api/animations/group) runs multiple animation steps in parallel.
```

**Review by petebacondarwin — APPROVED** [review]

**Pull request closed without merging**